compiler/rustc_expand/src/mbe/macro_parser.rs

```diff
@@ -587,9 +588,9 @@ impl TtParser {
         if *token == token::Eof {
             Some(match eof_mps {
                 EofMatcherPositions::One(mut eof_mp) => {
-                    // Need to take ownership of the matches from within the `Lrc`.
-                    Lrc::make_mut(&mut eof_mp.matches);
-                    let matches = Lrc::try_unwrap(eof_mp.matches).unwrap().into_iter();
+                    // Need to take ownership of the matches from within the `Rc`.
+                    Rc::make_mut(&mut eof_mp.matches);
+                    let matches = Rc::try_unwrap(eof_mp.matches).unwrap().into_iter();
```

Comment:
Would it make sense to just
```suggestion
                    let matches = Rc::make_mut(&mut eof_mp.matches).take().into_iter();
```

or even
```suggestion
                    eof_mp.matches.iter().cloned()
```